django_webapps/scripts/db_population/populate_teachers.py

```diff
@@ -38,25 +78,147 @@ def load_teachers(path_to_csv_file: str) -> list[list[str]]:
+            logger.info(f"Parsing teacher with tid {tid}")
+            self._current_teacher = teacher
+            self._create_teacher()
 
+    @transaction.atomic
     def _create_teacher(self) -> None:
-        pass
+        if self._current_teacher is None:
+            raise ValueError("Current teacher can not be None to call")
+
+        status = self._parse_column("status", teacher_parsers.parse_status)
+
+        tid = int(self._current_teacher[COLUMN_TO_ID["tid"]])
+
+        if (
+            status is not None
+            and status[0] == TeacherSituationalStatus.NO_RESPONSE
+            and tid > MAX_TID_WITH_NO_RESPONSE
+        ):
+            logger.info("Skipping teacher since no response from him")
+            return
+        none_teaching_help_comment = self._current_teacher[COLUMN_TO_ID["other_help_comment"]]
+        has_experience = self._parse_column("experience", teacher_parsers.parse_has_experience)
+        can_give_feedback = self._parse_column(
+            "can_give_feedback", teacher_parsers.parse_can_give_feedback
+        )
+        groups_number = self._parse_column("groups_number", teacher_parsers.parse_groups_number)
+        speaking_club = self._parse_column("speaking_club", teacher_parsers.parse_speaking_club)
+        personal_info = self._create_personal_info()
+        availability_slots = self._create_availability_slots()
+        languages_and_levels = self._create_language_and_levels()
+        age_ranges = self._create_age_ranges()
+        try:
+            teacher = Teacher.objects.create(
+                project_status=status[0] if type(status[0]) == TeacherProjectStatus else None,
+                has_prior_teaching_experience=has_experience or False,
+                peer_support_can_give_feedback=can_give_feedback or False,
+                simultaneous_groups=groups_number or 1,
+                can_host_speaking_club=speaking_club or False,
+                personal_info=personal_info,
+                status_since=timezone.now(),
+                is_validated=False,
+                weekly_frequency_per_group=1,
+                non_teaching_help_provided_comment=none_teaching_help_comment,
+                has_hosted_speaking_club=status[1]
+                == teacher_parsers.ProjectStatusAction.SPEAKING_CLUB,
+            )
+            teacher.student_age_ranges.set(age_ranges)
+            teacher.availability_slots.set(availability_slots)
+            teacher.teaching_languages_and_levels.set(languages_and_levels)
+            teacher.save()
+            logger.info(f"Successfully Parsed Teacher with tid {tid}")
+        except Exception as e:
+            self._report_error(f"Teacher with tid {tid} can not be parsed, see errors above, {e}")
+
+    def _report_error(self, error: ValueError) -> None:
+        logger.error(error)
+
+    def _parse_column(
+        self, column_name: str, parser: Callable[..., ParseColumnReturnType]
```

Comment:
Does the parser type not depend on the column name?
Maybe we should create an enumeration of column names and just plug in the correct parser right in this method? Otherwise it could look strange: "parse email with email parser" etc.

Reply:
Not all, we have days of week, since we call it only once I think it is fine for now